From: Sven Roederer <devel-sven@geroedel.de>
Date: Mon, 27 Apr 2020 02:22:35 +0200
Subject: tools/mkimage: update 110-fix-handling-of-long-pathnames.patch

Reapply on the updated version with the code that has been sent
upstream.

diff --git a/tools/mkimage/patches/110-fix-handling-of-long-pathnames.patch b/tools/mkimage/patches/110-fix-handling-of-long-pathnames.patch
index 9b7febd489c3415cccdf8e92ba8f4eede8c775a7..6f568858f5f7b854df8adfe765d291cd48448756 100644
--- a/tools/mkimage/patches/110-fix-handling-of-long-pathnames.patch
+++ b/tools/mkimage/patches/110-fix-handling-of-long-pathnames.patch
@@ -1,32 +1,74 @@
-diff -ur u-boot-2018.03.orig/tools/fit_image.c u-boot-2018.03/tools/fit_image.c
---- u-boot-2018.03.orig/tools/fit_image.c	2018-03-13 13:02:19.000000000 +0100
-+++ u-boot-2018.03/tools/fit_image.c	2019-10-24 17:16:23.914656651 +0200
-@@ -19,6 +19,7 @@
+From d7ed7b8f24ad970d2515a68d05dabfd6c1c51cee Mon Sep 17 00:00:00 2001
+From: Sven Roederer <devel-sven@geroedel.de>
+Date: Thu, 24 Oct 2019 16:29:05 +0200
+Subject: [PATCH 1/2] tools/mkimage: fix handling long filenames
+
+The cmdline for calling the dtc was cut-off when using long filenames (e.g.
+245 bytes) for output-file and datafile of "-f" parameter.
+For FIT-images cmd[MKIMAGE_MAX_DTC_CMDLINE_LEN] is declared (hardcoded 512 bytes),
+and contains some static values, the path of a tmpfile and a datafile. tmpfile is
+max MKIMAGE_MAX_TMPFILE_LEN (256) and datafile might be also this size. Having two
+very long pathname results in a truncation os the executed shell command, as the
+truncated datafile path will not be found.
+Redefine MKIMAGE_MAX_DTC_CMDLINE_LEN to "2 * MKIMAGE_MAX_TMPFILE_LEN + 35 for the
+parameters.
+This likely applies to the "-d" parameter, too.
+
+Signed-off-by: Sven Roederer <devel-sven@geroedel.de>
+---
+ tools/mkimage.h | 3 ++-
+ 1 file changed, 2 insertions(+), 1 deletion(-)
+
+diff --git a/tools/mkimage.h b/tools/mkimage.h
+index 0254af59fb..9eb18f4239 100644
+--- a/tools/mkimage.h
++++ b/tools/mkimage.h
+@@ -42,6 +42,7 @@ static inline ulong map_to_sysmem(void *ptr)
+ #define MKIMAGE_TMPFILE_SUFFIX		".tmp"
+ #define MKIMAGE_MAX_TMPFILE_LEN		256
+ #define MKIMAGE_DEFAULT_DTC_OPTIONS	"-I dts -O dtb -p 500"
+-#define MKIMAGE_MAX_DTC_CMDLINE_LEN	512
++#define MKIMAGE_MAX_DTC_CMDLINE_LEN	2 * MKIMAGE_MAX_TMPFILE_LEN + 35 
++/* 35 bytes for quotes, spaces, "-o", len(MKIMAGE_DEFAULT_DTC_OPTIONS), len(MKIMAGE_TMPFILE_SUFFIX) */
+ 
+ #endif /* _MKIIMAGE_H_ */
+-- 
+2.20.1
+
+
+From 3ec8cb222da62dfa785c0097cd981eee48b5baa3 Mon Sep 17 00:00:00 2001
+From: Sven Roederer <devel-sven@geroedel.de>
+Date: Thu, 24 Oct 2019 16:55:35 +0200
+Subject: [PATCH 2/2] tools/fit-image: print a warning when cmd-line for dtc
+ might be truncated
+
+Signed-off-by: Sven Roederer <devel-sven@geroedel.de>
+---
+ tools/fit_image.c | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/tools/fit_image.c b/tools/fit_image.c
+index 4aeabbcfe9..88ff093d05 100644
+--- a/tools/fit_image.c
++++ b/tools/fit_image.c
+@@ -17,6 +17,7 @@
+ #include "fit_common.h"
  #include "mkimage.h"
  #include <image.h>
- #include <stdarg.h>
 +#include <string.h>
+ #include <stdarg.h>
  #include <version.h>
  #include <u-boot/crc.h>
- 
-@@ -658,6 +659,9 @@
+@@ -744,6 +745,9 @@ static int fit_handle_file(struct image_tool_params *params)
  		snprintf(cmd, sizeof(cmd), "cp \"%s\" \"%s\"",
  			 params->imagefile, tmpfile);
  	}
 +	if (strlen(cmd) >= MKIMAGE_MAX_DTC_CMDLINE_LEN - 1) {
-+		fprintf(stderr, "WARNING: command-line for FIT creation might be truncated.\n");
++		fprintf(stderr, "WARNING: command-line for FIT creation might be truncated and will probably fail.\n");
 +	}
+ 
  	if (*cmd && system(cmd) == -1) {
  		fprintf (stderr, "%s: system(%s) failed: %s\n",
- 				params->cmdname, cmd, strerror(errno));
-diff -ur u-boot-2018.03.orig/tools/mkimage.h u-boot-2018.03/tools/mkimage.h
---- u-boot-2018.03.orig/tools/mkimage.h	2019-10-24 16:07:20.643345838 +0200
-+++ u-boot-2018.03/tools/mkimage.h	2019-10-24 17:27:38.682921690 +0200
-@@ -43,6 +43,6 @@
- #define MKIMAGE_TMPFILE_SUFFIX		".tmp"
- #define MKIMAGE_MAX_TMPFILE_LEN		256
- #define MKIMAGE_DEFAULT_DTC_OPTIONS	"-I dts -O dtb -p 500"
--#define MKIMAGE_MAX_DTC_CMDLINE_LEN	512
-+#define MKIMAGE_MAX_DTC_CMDLINE_LEN	2 * MKIMAGE_MAX_TMPFILE_LEN + 35 /* 35 bytes for quotes, space, -o, len(MKIMAGE_DEFAULT_DTC_OPTIONS), len(MKIMAGE_TMPFILE_SUFFIX) */
- 
- #endif /* _MKIIMAGE_H_ */
+-- 
+2.20.1
+
